lib/engine/game/g_18_usa/step/scrap_train.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative 'scrap_train_module'
+module Engine
+  module Game
+    module G18USA
+      module Step
+        class ScrapTrain < Engine::Step::Base
+          include ScrapTrainModule
+          def actions(entity)
+            actions = []
+            actions << 'scrap_train' if entity == current_entity && entity.corporation? &&
+                entity.trains.any? { |t| @game.pullman_train?(t) }
+            actions << 'pass' if blocks?
+            actions
+          end
+
+          def blocks?
+            @round.paid_loans[current_entity] && can_scrap_train?(current_entity)
+          end
+
+          def description
+            'Scrap Pullman'
+          end
+
+          def pass_description
+            'Skip Scrap Pullman'
+          end
+
+          def can_scrap_train?(entity)
+            return false unless entity.corporation?
+            return false unless entity.owned_by?(current_entity)
```

Comment:
What is this conditional doing? If it's the corporation's turn, won't this eject early?

Reply:
If you remove blocking? and let the loan step do that, you can remove this check as the actions method will gate the calls for you.

Reply:
scrap_train.rb is removed in the latest version of the PR